src/applications/edu-benefits/feedback-tool/helpers.js

```diff
@@ -251,7 +272,41 @@ export function transformSearchToolAddress({ address1, address2, address3, city,
     state: state && state.slice(0, searchToolSchoolAddressFields.state.maxLength),
     postalCode: zip && zip.slice(0, searchToolSchoolAddressFields.postalCode.maxLength),
   };
+}
 
+function addPrefilledFlagsToFormData(formData) {
+  const newFormData = { ...formData };
+  Object.keys(prefillFlagsToFieldsMap).forEach(flag => {
+    prefillFlagsToFieldsMap[flag].forEach(field => {
+      // if the value of this field is set on formData, add the flag to
+      // newFormData
+      if (get(field, formData)) {
+        newFormData[flag] = true;
+      }
+    });
+  });
+  return newFormData;
+}
+
+export function prefillTransformer(pages, formData, metadata) {
+  const newFormData = addPrefilledFlagsToFormData(formData);
+  return { metadata, formData: newFormData, pages };
+}
+
+/**
+ * Checks if the `prefillFlag` is set on the `data.formData`. If it is, returns
+ * the React Component created by the `messageComponentCreator`
+ *
+ * @param {string} prefillFlag - The key to look for on the form data
+ * @param {Object} data - Data object from form config
+ * @param {React Component} messageComponent - The React component to render
+ * @returns {React Component|null}
+ */
+export function conditionalPrefillMessage(prefillFlag, data, messageComponent) {
```

Comment:
maybe `maybeShowPrefillMessage` or something similar that expresses what this does.

Reply:
`conditionallyShowPrefillMessage`?